app/containers/Itunes Provider/TrackDetailsContainer/index.js

```diff
@@ -0,0 +1,164 @@
+import React, { memo, useState, useEffect } from 'react';
+import { compose } from 'redux';
+import PropTypes from 'prop-types';
+import { Skeleton, Card } from 'antd';
+import { useParams } from 'react-router-dom';
+import { get, isEmpty } from 'lodash';
+import styled from 'styled-components';
+import { connect } from 'react-redux';
+import { injectSaga } from 'redux-injectors';
+import { injectIntl } from 'react-intl';
+import { createStructuredSelector } from 'reselect';
+import { selectTrackErrors, selectSearchedTrack } from '../selectors';
+import If from '@components/If';
+import T from '@components/T';
+import For from '@components/For';
+import { searchContainerCreators } from '../reducer';
+import TrackDetailsComponent from '@app/components/TrackDetailsComponent/index';
+import searchContainerSaga from '../saga';
+
+const CustomCard = styled(Card)`
+  && {
+    margin: 20px 0;
+    max-width: 1000;
+    border-style: hidden;
+    background-color: #231f20;
+    height: 100vh;
+  }
+`;
+
+const Container = styled.div`
+  && {
+    display: flex;
+    flex-direction: column;
+    max-width: 1000;
+    width: 100%;
+    margin: 0 auto;
+    background-color: #231f20;
+    height: 100vh;
+    padding: ${(props) => props.padding}px;
+  }
+`;
+export function TrackDetailsContainer({ trackErrors, searchedTrack, dispatchGetTrackById }) {
+  const { trackId } = useParams();
+  const [loading, setLoading] = useState(false);
+  const [currentTrack, setCurrentTrack] = useState([]);
+
+  useEffect(() => {
+    setLoading(true);
+    dispatchGetTrackById(trackId);
+  }, [trackId]);
+
+  useEffect(() => {
+    const result = get(searchedTrack, 'results', []);
+    setCurrentTrack(result);
+    setLoading(false);
+  }, [searchedTrack]);
+
+  const renderTrackDetails = () => {
+    return (
+      <If condition={!isEmpty(currentTrack) || loading}>
+        <Container>
+          <CustomCard>
+            <Skeleton loading={loading} active>
+              <For
```

Comment:
do we need a For component here? We shouldn't be looping since we're dealing with a single entity right?

Reply:
semantically it doesn't really make sense